test/tracker-component.js

```diff
@@ -0,0 +1,148 @@
+/* globals describe before it */
+/* @flow */
+
+import { expect } from 'chai';
+
+import { Tracker } from '../src/tracker-component';
+
+describe('paypal.Tracker', () => {
+    let appendChildCalls = 0;
+    const appendChild = () => {
+        appendChildCalls += 1;
+    };
+    before(() => {
+        // $FlowFixMe
+        document.body = document.body || {};
+        // $FlowFixMe
+        document.body.appendChild = appendChild;
+    });
+
+    it('should be a function that returns a tracker', () => {
+        const userID = '__test__userID';
+        const userName = '__test__userName';
+        const tracker = Tracker({ user: { id: userID, name: userName } });
+        expect(tracker).to.have.property('view');
+        expect(tracker).to.have.property('addToCart');
+        expect(tracker).to.have.property('setCart');
+        expect(tracker).to.have.property('removeFromCart');
+        expect(tracker).to.have.property('purchase');
+        expect(tracker).to.not.have.property('purchases');
+    });
+
+    it('should send view events', () => {
+        const userID = '__test__userID2';
+        const userName = '__test__userName2';
+        const tracker = Tracker({ user: { id: userID, name: userName } });
+        expect(appendChildCalls).to.equal(0);
+        tracker.view({
+            pageUrl: 'https://example.com/test2'
+        });
+        expect(appendChildCalls).to.equal(1);
+    });
+
+    it('should send addToCart events', () => {
+        const userID = '__test__userID3';
+        const userName = '__test__userName3';
+        const tracker = Tracker({ user: { id: userID, name: userName } });
+        expect(appendChildCalls).to.equal(1);
+        tracker.addToCart({
+            cartId: '__test__cartId',
+            items:  [
+                {
+                    id:  '__test__productId',
+                    url: 'https://example.com/__test__productId'
+                }
+            ],
+            emailCampaignId: '__test__emailCampaignId',
+            price:           12345.67,
+            currencyCode:    'USD',
+            keywords:        [ '__test__' ]
+        });
+        expect(appendChildCalls).to.equal(2);
```

Comment:
Kind of an anti-pattern here for unit tests to depend on the state of the previous tests (i.e. instead of just incrementing the `appendChildCalls` for each test, increment it once per test and add an `afterEach` that resets it back to its original state). As it is, this particular test suite is kind of brittle, because if you were to change one of these tests or remove it, then all the other tests would break. This is a moot point though if you instead change your assertions to check the img.src instead of just checking that `appendChild` was called.

Reply:
Done